lib/datadog/tracing/distributed/ext.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+# typed: true
+
+module Datadog
+  module Tracing
+    module Distributed
+      # Cross-language keys used for distributed tracing.
+      # DEV-2.0: These constants are part of the public API through {Datadog::Tracing::Distributed::Headers::Ext}.
+      # DEV-2.0: We should not expose these constants, as we might not move them during refactor, and they are easily
+      # DEV-2.0: and publicly documented in Datadog's and B3's documentation.
+      module Ext
+        HTTP_HEADER_TRACE_ID = 'x-datadog-trace-id'
+        HTTP_HEADER_PARENT_ID = 'x-datadog-parent-id'
+        HTTP_HEADER_SAMPLING_PRIORITY = 'x-datadog-sampling-priority'
+        HTTP_HEADER_ORIGIN = 'x-datadog-origin'
+        # Distributed trace-level tags
+        HTTP_HEADER_TAGS = 'x-datadog-tags'
+
+        # Prefix used by all Datadog-specific distributed tags
+        DATADOG_PREFIX = 'x-datadog-'
+
+        # B3 keys used for distributed tracing.
+        # @see https://github.com/openzipkin/b3-propagation
+        B3_HEADER_TRACE_ID = 'x-b3-traceid'
+        B3_HEADER_SPAN_ID = 'x-b3-spanid'
+        B3_HEADER_SAMPLED = 'x-b3-sampled'
+        B3_HEADER_SINGLE = 'b3'
```

Comment:
Should these be moved to the B3 classes? They only seem to be used there.

Reply:
Yes, I have this comment at the top of this file, and in my opinion we don't need these to be part of the public API:
```
      # DEV-2.0: These constants are part of the public API through {Datadog::Tracing::Distributed::Headers::Ext}.
      # DEV-2.0: We should not expose these constants, as we might not move them during refactor, and they are easily
      # DEV-2.0: and publicly documented in Datadog's and B3's documentation.
```
I'll add that we should move B3 and other close to their usage location.

Reply:
If the only reason we're not moving them to the places we'd like to is backwards compatibility, have you considered making a copy instead?

E.g. leaving the original definitions in `Datadog::Tracing::Distributed::Headers::Ext` with a note saying "these should not be used and will be removed", and then add a local copy to the B3/etc classes that is used by those classes instead.

That way, we wouldn't need to do any further changes for 2.0 other than deleting `lib/datadog/tracing/distributed/headers/ext.rb`

Reply:
I can do that!

Reply:
Done, it looks much nicer, thank you!